ui/components/app/signature-request/signature-request.component.js

```diff
@@ -117,15 +118,16 @@ export default class SignatureRequest extends PureComponent {
           </div>
           <div className="signature-request-content__identicon-container">
             <div className="signature-request-content__identicon-initial">
-              {domain.name && domain.name[0]}
+              {/* {domain.name && domain.name[0]} */}
+              {originHostNameCharacter}
```

Comment:
In old component `subjectMetadata` is used to get icon site, https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/signature-request-original/signature-request-original.component.js#L139

Somehow, for v3/v4 requests `subjectMetadata` is not available in state.

Reply:
Sounds like we have to ensure this method is passing subjectMetadata with its invocation. This might be in the inpage script.

Reply:
I was playing around with this fix and it looked like this on a site. May be we should remove icon:

<img width="436" alt="Screenshot 2022-02-22 at 1 16 12 AM" src="https://user-images.githubusercontent.com/2182307/155017891-4f253916-3e4d-4a9c-9e3c-86c8c3199725.png">

Reply:
I could not find this validation but I have changed code in this component to use icon from `subjectMetadata` is available.

I am not sure if I should also use `subjectMetadata.name` over `origin` if available ?

Reply:
We stopped using `subjectMetadata.name` in favor of just using the `origin` because the name can be set to anything (so it can't be trusted), and because in practice it was typically set to something that didn't make for a good title (because we get it from the site's title, and page titles are apparently often full of junk like ads, notifications, etc.).